src/ngx_http_lua_headers_in.c

```diff
@@ -698,6 +693,11 @@ ngx_http_lua_set_input_header(ngx_http_request_t *r, ngx_str_t key,
     }
 #endif
 
+    if (r->headers_out.status == 400 || r->headers_in.headers.last == NULL) {
```

Comment:
Makes sense. Maybe the comment could become `must be a 400 / Bad Request` or something. Not very picky about this 😄